api/src/controllers/login.js

```diff
@@ -300,15 +314,27 @@ const renderLogin = (req) => {
   return render('login', req);
 };
 
+const renderPasswordReset = (req) => {
+  return render('passwordReset', req);
+}
+
 const login = async (req, res) => {
   try {
     const sessionRes = await createSession(req);
     if (sessionRes.statusCode !== 200) {
       res.status(sessionRes.statusCode).json({ error: 'Not logged in' });
-    } else {
-      const redirectUrl = await setCookies(req, res, sessionRes);
-      res.status(302).send(redirectUrl);
     }
+    const sessionCookie = getSessionCookie(sessionRes);
+    const options = { headers: { Cookie: sessionCookie } };
+    const userCtx = await getUserCtxRetry(options);
+    await setCookies(req, res, sessionRes);
```

Comment:
I think this is a bug. If you create the session and set the cookie _before_ they change their password then it's trivial to bypass the reset. I haven't tried this but if you go through this flow, and when you get to the "password reset" form, you just refresh the page, I think you'll be redirected right into the app!

Reply:
Yes that's true, this was a quick work around to just see the redirect happen and see the form. So that I can now design the form 😄